src/api/fungibleAsset.ts

```diff
@@ -42,10 +43,31 @@ export class FungibleAsset {
       pagination?: PaginationArgs;
       where?: FungibleAssetMetadataBoolExp;
     };
-  }): Promise<GetFungibleAssetMetadataResponse> {
+  }): Promise<GetFungibleAssetMetadatasResponse> {
     return getFungibleAssetMetadata({ aptosConfig: this.config, ...args });
   }
 
+  /**
+   * Queries the current specific fungible asset metadata
+   *
+   * This query returns the fungible asset metadata for a specific fungible asset.
+   *
+   * @param assetType The asset type of the fungible asset
+   * @returns getFungibleAssetMetadata A list of fungible asset metadata
+   */
+  async getFungibleAssetMetadataByAssetType(assetType: string): Promise<GetFungibleAssetMetadataResponse> {
```

Comment:
does it return a list? if so, we should support `options.pagination`. 
I would also follow the existing format, and create a `getFungibleAssetMetadataByAssetType` in `internal/fungibleAsset`

Reply:
So if we query by `asset type`, it should be 1 single item of the metadata. Noticed I added a new type `GetFungibleAssetMetadataResponse` for this case, and updated the list one as `GetFungibleAssetMetadatasResponse`. (The letter "s" might be hard to see haha)

And I was thinking on how to handle this. Since we already have a general query that can take an `where` condition in /internal. I think we can take advantage of that, and have many of the /api query depends on that one.
For example:

```
/internal
getFungibleAssetMetadata

/api
getFungibleAssetMetadata - same as above
getFungibleAssetMetadataByAssetType - calling the above with a `where` clause (_eq: asset_type)
getFungibleAssetMetadataByOwner - calling the above with a `where` clause (_eq: owner_address)
```

Any thoughts?

Reply:
then lets update the function comment `etFungibleAssetMetadata A list of fungible asset metadata`.

And yes we should definitely take advantage on the existing query. I still think it makes sense to have on the user facing api function `options` that holds `where` condition so they can filter on other things other than the asset type, then on the internal function, we should respect their custom filter and concat the assetType filer

Reply:
What is assetType here? FungibleAsset only goes by address while coin goes by the full struct tag. If this is only meant for fungible asset, this should always be an address

Reply:
@movekevin Yep you got it. This one is interesting, because it is being used interchangeably. 

"asset_type": "0x1::aptos_coin::AptosCoin" 
"asset_type": "0xc2948283c2ce03aafbb294821de7ee684b06116bb378ab614fa2de07a99355a8"

Both of the above are valid.

Reply:
@0xmaayan 

Hmm, but I think the asset_type is unique, so there is no need to do additional filter?